docs/core/diagnostics/snippets/resource-monitoring/Program.cs

```diff
@@ -0,0 +1,84 @@
+﻿// <setup>
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.ResourceMonitoring;
+using Microsoft.Extensions.Logging;
+using Spectre.Console;
+
+var services = new ServiceCollection()
+    .AddLogging(static builder => builder.AddConsole())
+    .AddResourceMonitoring();
+
+var provider = services.BuildServiceProvider();
+
+var monitor = provider.GetRequiredService<IResourceMonitor>();
+// </setup>
+
+// <monitor>
+await StartMonitoringAsync(monitor);
+
+static async Task StartMonitoringAsync(IResourceMonitor monitor)
```

Comment:
Ideally, to an async method we should pass CancellationToken. I think it would be nice to teach reader this pattern.